### PR TITLE
chore(observability): reduce cloning for events

### PR DESCRIPTION
**Pull request by jdrouet**
Following @bruceg suggestions from https://github.com/vectordotdev/vector/issues/9593

### CODE REVIEW

File: src/sources/aws_sqs/events.rs [hunk 1]
```diff
@@ -8,15 +8,12 @@ pub struct AwsSqsBytesReceived {
 }
 
 impl InternalEvent for AwsSqsBytesReceived {
```
Comment (bruceg) [inline src/sources/aws_sqs/events.rs:10]: Not related to your PR, but I'm curious why these are here and not in `src/internal_events`.

Comment (jszwedko) [reply src/sources/aws_sqs/events.rs:10]: Hmm, yeah, that is odd.

File: lib/vector-common/src/internal_event/mod.rs [hunk 2]
```diff
@@ -17,36 +16,35 @@ pub trait InternalEvent {
...
-    event.emit_logs();
-    event.emit_metrics();
-    if let Some(name) = event.name() {
-        super::event_test_util::record_internal_event(name);
+pub fn emit(event: impl InternalEvent) {
+    if let Some(name) = event.name().map(ToString::to_string) {
+        event.emit();
+        super::event_test_util::record_internal_event(name.as_str());
+    } else {
+        event.emit();
```
[the remark this answers is not in the document]
Comment (jdrouet) [reply lib/vector-common/src/internal_event/mod.rs]: done

File: src/sources/aws_sqs/events.rs (hunk 1 above)
Comment (jdrouet) [reply src/sources/aws_sqs/events.rs:10]: to be honest, I know it's the way to put them all in `src/internal_events`, but considering it's only used in this file, I don't see that as a problem 🤔

Comment (jszwedko) [reply src/sources/aws_sqs/events.rs:10]: I'd still move them over. We have other examples of events that are only used in one place that are in `src/internal_events` and it's nice to be able to just look there for event definitions.

**Review by jszwedko — APPROVED** [review]
Nice!

**Pull request merged**